consensus/src/rand/rand_gen/rand_manager.rs

```diff
@@ -261,10 +268,14 @@ impl<
                         response_sender,
                     } = request;
                     match rand_gen_msg {
+                        RandMessage::RequestShare(request) => {
+                            if let Some(share) = self.rand_store.lock().get_self_share(request.rand_metadata()) {
+                                self.process_response(protocol, response_sender, RandMessage::Share(share));
+                            }
+                        }
                         RandMessage::Share(share) => {
-                            match self.rand_store.add_share(share) {
-                                Ok(share_ack) => self.process_response(protocol, response_sender, RandMessage::ShareAck(share_ack)),
-                                Err(e) => error!("[RandManager] Failed to add share: {}", e),
+                            if let Err(e) = self.rand_store.lock().add_share(share) {
```

Comment:
do you also want to check if you can aggregate enough shares here?

Reply:
inside `add_share` it'll try to aggregate, after processing message there's a try_dequeue for each loop run